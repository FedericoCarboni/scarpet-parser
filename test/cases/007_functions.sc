// Function calls and definitions
func(param1, param2, outer(outer_param)) -> (
    null
);

// optional comma ,
func(1, 2,);
