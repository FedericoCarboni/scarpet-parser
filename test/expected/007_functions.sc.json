{
    "errors": [],
    "warnings": [],
    "root": {
        "kind": "BinaryExpression",
        "start": {
            "pos": 34,
            "row": 1,
            "col": 0
        },
        "end": {
            "pos": 125,
            "row": 6,
            "col": 12
        },
        "operator": ";",
        "lvalue": {
            "kind": "BinaryExpression",
            "start": {
                "pos": 34,
                "row": 1,
                "col": 0
            },
            "end": {
                "pos": 124,
                "row": 6,
                "col": 11
            },
            "operator": ";",
            "lvalue": {
                "kind": "BinaryExpression",
                "start": {
                    "pos": 34,
                    "row": 1,
                    "col": 0
                },
                "end": {
                    "pos": 90,
                    "row": 3,
                    "col": 1
                },
                "operator": "->",
                "lvalue": {
                    "kind": "FunctionExpression",
                    "start": {
                        "pos": 34,
                        "row": 1,
                        "col": 0
                    },
                    "end": {
                        "pos": 74,
                        "row": 1,
                        "col": 40
                    },
                    "name": "func",
                    "params": [
                        {
                            "kind": "Variable",
                            "start": {
                                "pos": 39,
                                "row": 1,
                                "col": 5
                            },
                            "end": {
                                "pos": 45,
                                "row": 1,
                                "col": 11
                            },
                            "name": "param1"
                        },
                        {
                            "kind": "Variable",
                            "start": {
                                "pos": 47,
                                "row": 1,
                                "col": 13
                            },
                            "end": {
                                "pos": 53,
                                "row": 1,
                                "col": 19
                            },
                            "name": "param2"
                        },
                        {
                            "kind": "FunctionExpression",
                            "start": {
                                "pos": 55,
                                "row": 1,
                                "col": 21
                            },
                            "end": {
                                "pos": 73,
                                "row": 1,
                                "col": 39
                            },
                            "name": "outer",
                            "params": [
                                {
                                    "kind": "Variable",
                                    "start": {
                                        "pos": 61,
                                        "row": 1,
                                        "col": 27
                                    },
                                    "end": {
                                        "pos": 72,
                                        "row": 1,
                                        "col": 38
                                    },
                                    "name": "outer_param"
                                }
                            ]
                        }
                    ]
                },
                "rvalue": {
                    "kind": "ParenthesisedExpression",
                    "start": {
                        "pos": 78,
                        "row": 1,
                        "col": 44
                    },
                    "end": {
                        "pos": 90,
                        "row": 3,
                        "col": 1
                    },
                    "value": {
                        "kind": "Constant",
                        "start": {
                            "pos": 84,
                            "row": 2,
                            "col": 4
                        },
                        "end": {
                            "pos": 89,
                            "row": 3,
                            "col": 0
                        },
                        "name": "null"
                    }
                },
                "comment": "// Function calls and definitions\n"
            },
            "rvalue": {
                "kind": "FunctionExpression",
                "start": {
                    "pos": 113,
                    "row": 6,
                    "col": 0
                },
                "end": {
                    "pos": 124,
                    "row": 6,
                    "col": 11
                },
                "name": "func",
                "params": [
                    {
                        "kind": "NumberLiteral",
                        "start": {
                            "pos": 118,
                            "row": 6,
                            "col": 5
                        },
                        "end": {
                            "pos": 119,
                            "row": 6,
                            "col": 6
                        },
                        "value": "1"
                    },
                    {
                        "kind": "NumberLiteral",
                        "start": {
                            "pos": 121,
                            "row": 6,
                            "col": 8
                        },
                        "end": {
                            "pos": 122,
                            "row": 6,
                            "col": 9
                        },
                        "value": "2"
                    },
                    null
                ]
            },
            "comment": null
        },
        "rvalue": null,
        "comment": "// optional comma ,\n"
    }
}